__docs__/wonder-blocks-core/exports.use-latest-ref.stories.mdx

```diff
@@ -0,0 +1,95 @@
+import {useEffect, useState} from "react";
+import {Meta} from "@storybook/addon-docs";
+import {useLatestRef, View} from "@khanacademy/wonder-blocks-core";
+import Button from "@khanacademy/wonder-blocks-button";
+import {LabeledTextField} from "@khanacademy/wonder-blocks-form";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import Color from "@khanacademy/wonder-blocks-color";
+
+<Meta
+    title="Core / Exports / useLatestRef()"
+    parameters={{
+        chromatic: {
+            disableSnapshot: true,
+        },
+    }}
+/>
+
+# useLatestRef()
+
+```ts
+function useLatestRef<T>(value: T): React.RefObject<T>;
+```
+
+The `useLatestRef` hook returns a ref that always contains the `value` passed
+to the hook during the most recent render.
+
+It can be used to wrap a possibly-changing prop in a stable value that can
+be passed to `useEffect`'s dependency array without causing unnecessary
+re-renders. For example:
```

Comment:
**suggestion:** I wonder if it's worth adding more illustrative info about why this is a valuable alternative to `useMemo`; for example, it allows for a value to be used by an effect without having to retrigger the effect function when it changes (or the effect's "unmount" call).

Reply:
I added a bit of detail. I'm not sure I understand the comparison to `useMemo`. Can `useMemo` be used to solve this problem?

Reply:
No, but the normal way to track a changed object but only change it when appropriate is to `useMemo` on that object, then use the memoized value in the dependency array. This removes the need to do that - you could save a whole object in the ref and then close around that in the effect, and avoid needing to update the effect (or any other hook that uses deps, like `useCallback`) just because the value changed.

To put it more plainly, the first instinct for folks might be to use `useMemo` when this may be a better, simpler option if the value is an object to be used in a closure.

Reply:
Writing hooks is really hard to get right as it involves thinking through a lot of interactions to determine how things will behave, including how React will do things.